hudi-common/src/main/java/org/apache/hudi/common/util/ClusteringUtils.java

```diff
@@ -333,28 +335,31 @@ public static List<HoodieInstant> getPendingClusteringInstantTimes(HoodieTableMe
    * Returns the earliest instant to retain.
    * Make sure the clustering instant won't be archived before cleaned, and the earliest inflight clustering instant has a previous commit.
    *
-   * @param activeTimeline The active timeline
-   * @param metaClient     The meta client
+   * @param activeTimeline               The active timeline
+   * @param metaClient                   The meta client
+   * @param cleanerPolicy                The hoodie cleaning policy
+   * @param shouldArchiveBeyondSavepoint Is archival enabled to run beyond savepoint
    * @return the earliest instant to retain for clustering
    */
   public static Option<HoodieInstant> getEarliestInstantToRetainForClustering(
-      HoodieActiveTimeline activeTimeline, HoodieTableMetaClient metaClient) throws IOException {
+      HoodieActiveTimeline activeTimeline, HoodieTableMetaClient metaClient, HoodieCleaningPolicy cleanerPolicy, boolean shouldArchiveBeyondSavepoint) throws IOException {
     Option<HoodieInstant> oldestInstantToRetain = Option.empty();
     HoodieTimeline replaceOrClusterTimeline = activeTimeline.getTimelineOfActions(CollectionUtils.createSet(HoodieTimeline.REPLACE_COMMIT_ACTION, HoodieTimeline.CLUSTERING_ACTION));
+    Option<HoodieInstant> cleanInstantOpt =
```

Comment:
Why moves the `cleanInstantOpt` out from the check `!replaceOrClusterTimeline.empty()` ?

Reply:
Addressed